Blizzard.IO.Core/IConsumerWithMetadata.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace Blizzard.IO.Core
+{
+    public interface IConsumerWithMetadata<TData, TMetadata> where TMetadata : BaseMetadata
+    {
+        event Action<TData, TMetadata> MessageWithMetadataConsumed;
```

Comment:
I believe it should be `MessageReceived`

Reply:
Same as above